File: app/expense.py
from datetime import datetime


class Expense:
    def __init__(
        self,
        expense_id,
        description,
        amount,
        category,
        created_at=None
    ):
        self.id = expense_id
        self.description = description
        self.amount = amount
        self.category = category
        self.created_at = created_at or datetime.now()

File: frontend/frontend/src/types/expense.ts
// Backed by a real Python module today: app/expense_manager.py + app/database/repository.py
// Field names mirror the CSV export sample in the repo README (ID, Description, Amount, Category, Created At).
// Confirm exact field names/types against expense.py before wiring up FastAPI.

export interface Expense {
  id: number;
  description: string;
  amount: number;
  category: string;
  created_at: string; // ISO datetime string
}

export type NewExpense = Omit<Expense, 'id' | 'created_at'>;
export type UpdateExpense = Partial<NewExpense>;

export interface ExpenseFilters {
  search?: string;
  category?: string;
  minAmount?: number;
  maxAmount?: number;
}

export interface ExpenseCategoryTotal {
  category: string;
  total: number;
}
